std/range/package.d

```diff
@@ -10375,6 +10375,17 @@ private struct OnlyResult(T)
     }
     alias opDollar = length;
 
+    // FIXME Workaround for https://issues.dlang.org/show_bug.cgi?id=24415
+    import std.traits : hasElaborateCopyConstructor;
+    static if (hasElaborateCopyConstructor!T)
+    {
+        public this()(ref return scope inout(OnlyResult) src) inout
```

Comment:
[Copy constructors can't be templates][1], which means that this constructor overload is not actually being used as a copy constructor (and the compiler is still generating one).

Fortunately, the public overload doesn't have to *do anything* in order to successfully work around issue 24415, it just has to *exist*. So, maybe we should make it a dummy overload instead. E.g.,

```d
        private static struct WorkaroundBugzilla24415 {}
        public this(WorkaroundBugzilla24415) {}
```

[1]: https://dlang.org/spec/struct.html#struct-copy-constructor

Reply:
Oh, drat. I missed that part of the spec. Okay. I'll update it accordingly.

Reply:
Okay. For some reason, that solution didn't work unless I made the function a template, but it does work properly if it's a template (which does also have the benefit of avoiding adding another symbol). So, I pushed an updated fix.

That being said, if it's not a template, you get a bizarre situation where `OnlyResult` has a public copy constructor that was generated by the compiler, but when code tries to call it, the result is a compiler error about `OnlyResult` not having a `__ctor` member (or more specifically, it complains about the variable that's the copy not having a `__ctor` member). So, there's probably another compiler bug that needs to be reduced and reported there, but as long as the workaround constructor is templated, that bug doesn't manifest.